vshard/util.lua

```diff
@@ -21,6 +21,43 @@ end
 
 local tnt_version = lversion.parse(_TARANTOOL)
 
+--
+-- Deep comparison of tables. Stolen from built-in table.equals() which sadly is
+-- not available in 1.10.
+--
+local function table_equals(a, b)
+    if type(a) ~= 'table' or type(b) ~= 'table' then
+        return type(a) == type(b) and a == b
+    end
+    local mta = getmetatable(a)
+    local mtb = getmetatable(b)
+    if mta and mta.__eq or mtb and mtb.__eq then
+        return a == b
+    end
```

Comment:
nit: Not completely sure that you need (even in future) this metatable magic.

Reply:
I intentionally copy-pasted this from Tarantool master as is so as not to diverge too much from how it works in new versions. I might not needed it for configs but could need for something else in the future.